shcDecoder/src/main/java/ca/bc/gov/shcdecoder/SHCVerifierImpl.kt

```diff
@@ -105,28 +108,36 @@ class SHCVerifierImpl(
                     vaxCode?.toInt() == vaccineRule.cvxCode
                 }
 
-                when (ruleSet?.type) {
-                    1 -> {
-                        mrnType += ruleSet.ru
-                    }
+                val hasNumberOfDasElapsedSinceLastDose = intervalPassed(lastVaxDate, minInterval)
 
-                    2 -> {
-                        nrvvType += ruleSet.ru
-                    }
+                if (hasNumberOfDasElapsedSinceLastDose) {
```

Comment:
I think we should skip all of the rest of the code if this is false. In particular `lastVaxDate = vaxDate` shouldn't happen.

Reply:
it may happen if ru required = 3 and vax ru value = 1 for all the three-dose.
Also in the future, if a booster dose comes in this makes sense.